src/test/unit/services/sample/hmc_nuts_dense_mass_matrix_test.cpp

```diff
@@ -0,0 +1,264 @@
+#include <stan/services/sample/hmc_nuts_dense_e_adapt.hpp>
+#include <stan/services/sample/hmc_nuts_dense_e.hpp>
+#include <stan/io/empty_var_context.hpp>
+#include <test/test-models/good/mcmc/hmc/common/gauss3D.hpp>
+#include <test/unit/services/instrumented_callbacks.hpp>
+#include <test/unit/services/check_adaptation.hpp>
+#include <iostream>
+#include <gtest/gtest.h>
+
+/** 
+ * Use 3-param model test-models/good/mcmc/hmc/common/gauss3D
+ * fix seed 12345, test against specified inv mass matrix
+ * Tests crafted by running samplers with test config
+ * to capture resulting inverse mass matrix values.
+ */
+
+class ServicesSampleHmcNutsDenseEMassMatrix : public testing::Test {
+public:
+  ServicesSampleHmcNutsDenseEMassMatrix()
+    : model(context, &model_log) {}
+
+  std::stringstream model_log;
+  stan::test::unit::instrumented_writer message, init, error;
+  stan::test::unit::instrumented_writer parameter, diagnostic;
+  stan::io::empty_var_context context;
+  stan_model model;
+};
+
+TEST_F(ServicesSampleHmcNutsDenseEMassMatrix, ident_no_adapt) {
+  unsigned int random_seed = 12345;
+  unsigned int chain = 1;
+  double init_radius = 2;
+  int num_warmup = 0;
+  int num_samples = 2;
+  int num_thin = 1;
+  bool save_warmup = false;
+  int refresh = 0;
+  double stepsize = 1;
+  double stepsize_jitter = 0;
+  int max_depth = 10;
+  stan::test::unit::instrumented_interrupt interrupt;
+  EXPECT_EQ(interrupt.call_count(), 0);
+
+  int return_code =
+    stan::services::sample::hmc_nuts_dense_e(model,
+                                             context,
+                                             random_seed,
+                                             chain,
+                                             init_radius,
+                                             num_warmup,
+                                             num_samples,
+                                             num_thin,
+                                             save_warmup,
+                                             refresh,
+                                             stepsize,
+                                             stepsize_jitter,
+                                             max_depth,
+                                             interrupt,
+                                             message,
+                                             error,
+                                             init,
+                                             parameter,
+                                             diagnostic);
+  EXPECT_EQ(0, return_code);
+
+  stan::io::dump dmp = 
+    stan::services::util::create_ident_dense_mass_matrix(3);
+  stan::io::var_context& inv_mass_matrix = dmp;
+  std::vector<double> dense_vals
+    = inv_mass_matrix.vals_r("mass_matrix");
```

Comment:
Missing a `inv_` prefix of the name within the var context here and throughout the test?